src/components/Layout/UserControls.tsx

```diff
@@ -1,40 +1,98 @@
-  notifications: OwnerRequestType[];
-  logout: () => void;
+type UserControls = {
+  defaultTheme: string;
 };
 
-export const UserControls: React.FC<UserControlsProps> = ({
-  userId,
-  userName,
-  notifications,
-  logout,
-}) => (
-  <>
-    <Link
-      href={`${Routes.USER}/${userId}`}
-      className="relative flex items-center justify-center text-xl hover:text-primary focus:text-primary"
-      aria-label={`View ${userName}'s profile`}
-    >
-      <ProfileIcon
-        aria-label="User profile"
-        className="inline h-6 w-6 fill-text hover:fill-primary focus:fill-primary sm:hidden"
-      />
-      <span className="hidden whitespace-nowrap sm:inline">{userName}</span>
-      {notifications?.length > 0 && (
-        <span className="absolute -right-3 -top-1 flex min-h-5 min-w-5 items-center justify-center rounded-full bg-mandarin p-0.5 text-xs text-white">
-          {notifications.length > 9 ? '9+' : notifications.length}
-        </span>
-      )}
-    </Link>
-    <Button onClick={logout}>Odhlásit&nbsp;se</Button>
-  </>
-);
+export const UserControls: React.FC<UserControls> = ({ defaultTheme }) => {
+  const {
+    user: { id, name, email },
+    logout,
+  } = useAuth();
+
+  const [isNavigationOpen, setIsNavigationOpen] = useState(false);
+
+  const [notifications, setNotifications] = useState<OwnerRequestType[]>([]);
+
+  const { openAuthModal } = useAuthModal();
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const isUserLoggedIn = !!email;
+
+  useEffectAsync(async () => {
+    if (!isUserLoggedIn) {
+      return;
+    }
+
+    try {
+      const notificationsData = await getOwnerRequests({ isApproved: null });
+
+      if (notificationsData) {
+        setNotifications(notificationsData);
+      }
+    } catch (error) {
+      setNotifications([]);
+    }
+  }, []);
+
+  useClickOutside(ref, () => setIsNavigationOpen(false), isNavigationOpen);
+
+  const handleOpenNavigation = () => {
+    setIsNavigationOpen(!isNavigationOpen);
+  };
+
+  return (
+    <div ref={ref} className="flex items-center">
+      <button
+        className="items-center h-full justify-center text-xl hover:text-primary focus:text-primary relative"
+        onClick={isUserLoggedIn ? handleOpenNavigation : openAuthModal}
+      >
+        <div className="flex gap-4 items-center">
+          <span className="hidden whitespace-nowrap sm:inline">{name}</span>
+          <ProfileIcon className="inline h-6 w-6 fill-text hover:fill-primary focus:fill-primary" />
+        </div>
+
+        {notifications?.length > 0 && (
+          <span className="text-xs bg-mandarin text-white absolute font-semibold -top-1 -right-3 p-0.5 min-w-5 min-h-5 rounded-full flex justify-center items-center">
+            !
+          </span>
+        )}
+      </button>
+      <div
+        className={cx(
+          'absolute top-14 right-0 w-48 bg-background border border-low-contrast flex-col gap-3 rounded-md p-5',
+          isNavigationOpen ? 'flex' : 'hidden',
```

Comment:
What about this?

```suggestion
          'absolute top-14 right-0 w-48 bg-background border border-low-contrast flex flex-col gap-3 rounded-md p-5',
          { 'hidden' !isNavigationOpen },
```

or the opposite, without the negation